corehq/toggles/__init__.py

```diff
@@ -2603,9 +2603,13 @@ def _handle_attendance_tracking_role(domain, is_enabled):
 
 def _handle_geospatial_es_index(domain, is_enabled):
     from corehq.apps.geospatial.tasks import index_es_docs_with_location_props
+    from corehq.apps.geospatial.utils import get_celery_task_tracker
+    from corehq.apps.geospatial.const import ES_INDEX_TASK_HELPER_BASE_KEY
 
     if is_enabled:
-        index_es_docs_with_location_props.delay(domain)
+        celery_task_tracker = get_celery_task_tracker(domain, ES_INDEX_TASK_HELPER_BASE_KEY)
```

Comment:
I like that we were able to check this here, though, should not we fail the save if there is a task in progress?
That is what I was thinking when I originally requested that change.
Otherwise its doing the same thing, just in a different place.

Reply:
I'm on the fence about failing here since this doesn't completely feel like an error. If there is already a task in progress then the user will simply continue to see the progress of that task. This check simply enforces that each domain's task is handled in serial.

Though I do see that it could be helpful to know that this action was blocked. I'm just not sure where that notification should be shown. Should it be shown to the user on the Microplanning pages or on the edit FF page when enabling the Microplanning FF? The latter does look like a bigger lift since there currently seems to be no easy way to bubble up a specific message to the user on the edit FF page.

Reply:
> If there is already a task in progress then the user will simply continue to see the progress of that task

That makes sense to me, So we are just ignoring another request to re-index because one is already running.
That sits well with me. It just was not explicit to me when I was looking at the code. May be a comment in code would help anyone who would come across it.

Feel free to put the change back to where it originally was if you prefer that.

Reply:
I'll leave it here as this can help in avoiding firing unnecessary Celery tasks, however I'll add a comment here as suggested.

Reply:
Addressed in [6403610](https://github.com/dimagi/commcare-hq/pull/35126/commits/640361072b67b7ae6121a2c127cfc60d1298ca3d).